crates/shared/src/oneinch_api.rs

```diff
@@ -412,7 +483,7 @@ impl ProtocolCache {
         ))))
     }
 
-    pub async fn get_all_protocols(&self, api: &dyn OneInchClient) -> Result<Vec<String>> {
+    pub async fn get_all_protocols(&self, api: &dyn OneInchClient) -> Result<Vec<LiquidiySource>> {
```

Comment:
Not sure whether it's worth the effort, but you could rename the functions /structs now : `get_all_liquidity_sources` or `LiquiditySourceCache` are two examples. I think then it would be easier to maintain in the future.

Reply:
Great feedback! I'm a bit conflicted about this, though.
1Inch's naming scheme seems inconsistent so I believe whatever we end up naming the functions and types will never by 100% consistent as well.
For example the `/liquidity-sources` endpoint returns a list of `protocols`. But the items of this list are not identical to items in the `protocols` field of the `/swap` and `/quote` endpoints.
Right now the term `protocol` seems to be more common in the 1Inch language than `liquidity source`.
Maybe renaming `LiquiditySource -> Protocol` and `Protocol -> TradeRouteSegment` (or something similar) could be an option. This way the 1Inch naming inconsistencies don't leak into the user API.
Then the user will not have to wonder how they can convert a `LiquiditySource` to a thing they can pass as a `protocol` argument, when they are basically the same thing.

Reply:
I'd be slightly in favour of `get_liquidity_sources` just because that is what the URL is called (it makes it easier for me to match the 1Inch API client function to the endpoint it corresponds to).

As for the protocol names, I would call the struct `ProtocolInfo` just because its in a the `protocols` field of the returned JSON, but isn't a protocol name (as passed into the route). Also, we can rename the current `Protocol` struct to `ProtocolRouteSegment` or something (again, just because of my _weak_ argument that the field in the JSON block is `protocols`). For the protocol names, I really don't have a strong opinion. I'm also happy with your suggestions as well!